instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/webmvc/v6_0/DispatcherServletInstrumentation.java

```diff
@@ -61,13 +62,26 @@ public static class HandlerMappingAdvice {
     public static void afterRefresh(
         @Advice.Argument(0) ApplicationContext springCtx,
         @Advice.FieldValue("handlerMappings") List<HandlerMapping> handlerMappings) {
-      if (springCtx.containsBean("otelAutoDispatcherFilter")) {
-        OpenTelemetryHandlerMappingFilter filter =
-            (OpenTelemetryHandlerMappingFilter) springCtx.getBean("otelAutoDispatcherFilter");
-        if (handlerMappings != null) {
-          filter.setHandlerMappings(handlerMappings);
+
+      if (handlerMappings == null || !springCtx.containsBean("otelAutoDispatcherFilter")) {
+        return;
+      }
+
+      Object bean = springCtx.getBean("otelAutoDispatcherFilter");
+      OpenTelemetryHandlerMappingFilter filter;
+      if (bean instanceof OpenTelemetryHandlerMappingFilter) {
```

Comment:
[for reviewer] this is an edge-case where the inlined advice will be able to directly call the `OpenTelemetryHandlerMappingFilter` class directly and with indy we get a proxy instance that needs to be unwrapped (thus the `delegate` fields needs to be made public for simplicity).